src/com/etendoerp/sequences/SequencesGenerator.java

```diff
@@ -25,118 +26,128 @@
+          }
+        } else {
+          count = count + createSequence(client, org, name, column, null);
         }
-        return 0;
+      }
     }
-
-    public boolean existsSequence(Column column, Client client, Organization organization, DocumentType documentType){
-        OBCriteria<Sequence> sequenceOBCriteria = OBDal.getInstance().createCriteria(Sequence.class);
-        sequenceOBCriteria.add(Restrictions.eq(Sequence.PROPERTY_CLIENT, client));
-        sequenceOBCriteria.add(Restrictions.eq(Sequence.PROPERTY_COLUMN, column));
-        if(documentType!= null){
-            sequenceOBCriteria.add(Restrictions.eq(Sequence.PROPERTY_DOCUMENTTYPE, documentType));
-        }
-        sequenceOBCriteria.add(Restrictions.eq(Sequence.PROPERTY_ORGANIZATION, organization));
-        return !sequenceOBCriteria.list().isEmpty();
+    return count;
+  }
+
+  public boolean hasDocType(Table t) {
+    Entity entity = ModelProvider.getInstance().getEntity(t.getName());
+    return entity.hasProperty(SequenceDatabaseUtils.PROPERTY_DOCUMENTTYPE) || entity.hasProperty(
+        SequenceDatabaseUtils.PROPERTY_DOCUMENTTYPE_TARGET);
+  }
+
+  public int createSequence(Client client, Organization organization, String name, Column column,
+      DocumentType documentType) {
+    if (!existsSequence(column, client, organization, documentType)) {
+      setSequenceValues(client, organization, name, column, documentType);
+      return 1;
     }
-
-    @Override
-    protected Class<?> getInputClass() {
-        return Sequence.class;
+    return 0;
+  }
+
+  public Sequence setSequenceValues(Client client, Organization organization, String name, Column column,
+      DocumentType documentType) {
+    final Sequence sequence = OBProvider.getInstance().get(Sequence.class);
+    // set values
+    sequence.setClient(client);
+    sequence.setOrganization(organization);
+    sequence.setName(name);
+    sequence.setPrefix("");
+    sequence.setSuffix("");
+    sequence.setMask("#######");
+    sequence.setActive(true);
+    sequence.setColumn(column);
+    sequence.setTable(column.getTable());
+    sequence.setDocumentType(documentType);
+    sequence.setAutoNumbering(true);
+    sequence.setNextAssignedNumber(1000000L);
+    sequence.setIncrementBy(1L);
+
+    // store it in the database
+    OBDal.getInstance().save(sequence);
+    return sequence;
+  }
+
+  public boolean existsSequence(Column column, Client client, Organization organization, DocumentType documentType) {
+    OBCriteria<Sequence> sequenceOBCriteria = OBDal.getInstance().createCriteria(Sequence.class);
+    sequenceOBCriteria.add(Restrictions.eq(Sequence.PROPERTY_CLIENT, client));
+    sequenceOBCriteria.add(Restrictions.eq(Sequence.PROPERTY_COLUMN, column));
+    if (documentType != null) {
+      sequenceOBCriteria.add(Restrictions.eq(Sequence.PROPERTY_DOCUMENTTYPE, documentType));
     }
+    sequenceOBCriteria.add(Restrictions.eq(Sequence.PROPERTY_ORGANIZATION, organization));
```

Comment:
Es mas rapido usar un setMaxResult(1), un uniqueResult y preguntar si el resultado es distinto de null asi no usas toda la tabla

Reply:
The GitHub comments, should be in English